test/common/runtime/runtime_protos_test.cc

```diff
@@ -24,6 +24,25 @@ class RuntimeProtosTest : public testing::Test {
   NiceMock<MockLoader> runtime_;
 };
 
+TEST_F(RuntimeProtosTest, UInt32Test) {
+  envoy::config::core::v3::RuntimeUInt32 uint32_proto;
+  std::string yaml(R"EOF(
+runtime_key: "foo.bar"
+default_value: 99
+)EOF");
+  TestUtility::loadFromYamlAndValidate(yaml, uint32_proto);
+  UInt32 test_uint32(uint32_proto, runtime_);
+
+  EXPECT_CALL(runtime_.snapshot_, getInteger("foo.bar", 99));
+  EXPECT_EQ(99, test_uint32.value());
+
+  EXPECT_CALL(runtime_.snapshot_, getInteger("foo.bar", 99)).WillOnce(Return(1024));
+  EXPECT_EQ(1024, test_uint32.value());
+
+  EXPECT_CALL(runtime_.snapshot_, getInteger("foo.bar", 99)).WillOnce(Return(1ull << 33));
+  EXPECT_EQ(0, test_uint32.value());
```

Comment:
In cases where getInteger returns a value larger than uint32 max, should this mask the returned value down to 32 bits or return the default value?  Consistency between the behavior of getDouble and getInteger for values larger than uint64 max suggest that we should return the default in this case.

https://github.com/envoyproxy/envoy/blob/main/source/common/runtime/runtime_impl.h#L104

Reply:
The default value is indeed more reasonable, I'll submit a commit to fix it.

Reply:
Fixed in the latest commit. @antoniovicente